torch/optim/lr_scheduler.py

```diff
@@ -47,7 +48,8 @@ def step(self, epoch=None):
             epoch = self.last_epoch + 1
         self.last_epoch = epoch
         for param_group, lr in zip(self.optimizer.param_groups, self.get_lr()):
-            param_group['lr'] = lr
+            if self.min_lr is None or lr > self.min_lr:
+                param_group['lr'] = lr
```

Comment:
does that mean that `param_group['lr']` should not be updated when the update value is below `min_lr`, even though `param_group['lr']` is greater (possibly by far) than `min_lr` ? Say you start at `lr=1` with a minimum `min_lr=0.1` and try to update it with `lr=0.01` should it stay `1`or shoukd it become `0.1` ?

Isn't it better to saturate `lr` `min_lr` so that it will be the truly minimum value reached ?

```python
fi self.min_lr is None:
    param_group['lr'] = lr
else:
    param_group['lr'] = max(lr, self.min_lr)

```